XDMA/linux-kernel/libxdma/libxdma.c

```diff
@@ -2546,9 +2548,8 @@ static void xdma_desc_adjacent(struct xdma_desc *desc, int next_adjacent)
 		extra_adj = next_adjacent - 1;
 		if (extra_adj > MAX_EXTRA_ADJ)
 			extra_adj = MAX_EXTRA_ADJ;
-		max_adj_4k =
-			(0x1000 - ((le32_to_cpu(desc->next_lo)) & 0xFFF)) / 32 -
-			1;
+		max_adj_4k = (PAGE_SIZE_X86 - ((le32_to_cpu(desc->next_lo)) &
+			PAGE_MASK_X86)) / sizeof(struct xdma_desc) - 1;
```

Comment:
This macro is hardcore. I guess you all have moved on to aarch ;)

Reply:
I'm trying to make sure that the driver works on x86 and aarch architectures, which is why I made this change. The adjacency accesses make x86 page size assumptions, which is why propose adding these macros instead of just using "PAGE_SIZE" and "PAGE_MASK", which are standard, and much better than the magic numbers in the current code.